Sources/Angle.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  Angle.swift
+//  Euclid
+//
+//  Created by Ioannis Kaliakatsos on 22.11.20.
+//  Copyright © 2020 Nick Lockwood. All rights reserved.
+//
+
+import Foundation
+
+#if os(Linux)
+import Glibc
+#else
+import Darwin
+#endif
+
+/// A type-safe struct for all API related to angles
+public struct Angle {
+    public let degrees: Double
+
+    public init(degrees: Double) {
+        self.degrees = degrees
+    }
+}
+
+public extension Angle {
+    private static let radiansPerDegree = Double.pi / 180
+
+    internal init(radians: Double) {
+        self.init(degrees: radians / Angle.radiansPerDegree)
+    }
+
+    var radians: Double { degrees * Angle.radiansPerDegree }
+}
+
+public extension Angle {
+    static var zero = Angle(degrees: 0)
+
+    static var piHalf = Angle(degrees: 90)
+
+    static var pi = Angle(degrees: 180)
+
+    static var twoPi = Angle(degrees: 360)
+}
+
+public extension Angle {
+    var cos: Double { os.cos(radians) }
```

Comment:
Semantically I don't think these should be properties since they are applying a transformation.

Reply:
My intention was explicit, so they don't read like function calls but as properties. Similar to mathematics that a given angle _has_ a cosine etc.

However, you have the final word on this, so let me know how you'd like it.